### PR TITLE
[CHK-1129] Mask cvv pan email log

### PR DESCRIPTION
**Pull request by ciuffagianluca**
Mask data about cvv pan and email when they could be prionted in log message

#### List of Changes

Added logback configuration to handle custom layout message

#### Motivation and Context

we don't want to log sensitive data

#### How Has This Been Tested?

Tested in local in environment
Added junit test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/test/java/it/pagopa/transactions/utils/LogMaskTests.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                                new TransactionAmount(100),
+                                new TransactionDescription("description"),
+                                new PaymentContextCode(null)
+                        )
+                ),
+                null,
+                null,
+                null,
+                it.pagopa.ecommerce.commons.documents.v1.Transaction.ClientId.CHECKOUT
+        );
```
Comment (giovanniberti) [inline src/test/java/it/pagopa/transactions/utils/LogMaskTests.java:88]: Use `TransactionTestUtils.transactionActivated()` 🙂

Comment (ciuffagianluca) [reply src/test/java/it/pagopa/transactions/utils/LogMaskTests.java]: Funny!

File: src/main/java/it/pagopa/transactions/utils/LogMaskerPatternLayout.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.stream.IntStream;
+
+public class LogMaskerPatternLayout extends PatternLayout {
+    private Pattern patternString;
+    private List<String> maskPatterns = new ArrayList<>();
+
+    public void addMaskPattern(String maskPattern) {
+        maskPatterns.add(maskPattern);
+        patternString = Pattern.compile(maskPatterns.stream().collect(Collectors.joining("|")), Pattern.MULTILINE);
+    }
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/utils/LogMaskerPatternLayout.java:21]: Where will be this method be called at runtime to add the mask patterns?

Comment (ciuffagianluca) [reply src/main/java/it/pagopa/transactions/utils/LogMaskerPatternLayout.java:21]: It is not directly called by any other point in the project, but it is necessary to make the logback configuration file working, when it add the mask patterns to the list. If you try to comment that method and start the application, the start will fail.

**Pull request merged**